### PR TITLE
Add `STDERR.puts` check to `Style/StderrPuts` cop

### PR DESCRIPTION
**Pull request by elebow**
`STDERR` is often equal to `$stderr` in Ruby programs. Writing to either one should be replaced by `warn`.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by pocke** [comment]
link: https://github.com/rubocop-hq/rubocop/pull/4813#discussion_r142003554
We have discussed it, but I guess the new cop that is discussed in the link has not been implemented yet.

@koic what do you think about this pull-request?

File: spec/rubocop/cop/style/stderr_puts_spec.rb [hunk 1]
```diff
@@ -8,11 +8,11 @@
   it "registers an offense when using `$stderr.puts('hello')`" do
     expect_offense(<<-RUBY.strip_indent)
       $stderr.puts('hello')
-      ^^^^^^^^^^^^ Use `warn` instead of `$stderr.puts` to allow such output to be disabled.
+      ^^^^^^^^^^^^ Use `warn` instead of `$stderr.puts` or `STDERR.puts` to allow such output to be disabled.
```
Comment (jaredbeck) [inline spec/rubocop/cop/style/stderr_puts_spec.rb]: A very minor thing, but adding the word "either" here might help ("instead of either").

File: spec/rubocop/cop/style/stderr_puts_spec.rb [hunk 2]
```diff
@@ -23,4 +23,23 @@
...
+    new_source = autocorrect_source("STDERR.puts('hello')")
+
+    expect(new_source).to eq "warn('hello')"
+  end
+
+  it 'registers no offense when using `STDERR.puts` with no arguments' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      STDERR.puts
+    RUBY
+  end
```
Comment (jaredbeck) [inline spec/rubocop/cop/style/stderr_puts_spec.rb]: I believe that `STDERR.puts` with no arguments *should* register an offense. Printing a newline should be no different that printing anything else. Perhaps this should be discussed in a separate PR, though?

File: spec/rubocop/cop/style/stderr_puts_spec.rb (hunk 1 above)
Comment (elebow) [reply spec/rubocop/cop/style/stderr_puts_spec.rb]: Thanks. Text updated.

File: spec/rubocop/cop/style/stderr_puts_spec.rb (hunk 2 above)
Comment (elebow) [reply spec/rubocop/cop/style/stderr_puts_spec.rb]: Agreed on both counts. I can't think of a situation where it would be useful to allow `STDERR.puts` without arguments but prohibit it with arguments. But for this PR, I think it's best to follow the example of the legacy logic.

File: lib/rubocop/cop/style/stderr_puts.rb [hunk 3]
```diff
@@ -17,12 +17,13 @@ module Style
       class StderrPuts < Cop
         include RangeHelp
 
-        MSG = 'Use `warn` instead of `$stderr.puts` to allow such output ' \
-              'to be disabled.'.freeze
+        MSG = 'Use `warn` instead of either `$stderr.puts` or `STDERR.puts` ' \
```
Comment (bbatsov) [inline lib/rubocop/cop/style/stderr_puts.rb]: I think in the message we should show exactly what we detected (meaning we should make the message dynamic).

**Comment by elebow** [comment]
Updated to address comments.

**Comment by bbatsov** [comment]
Thanks!
